iceoryx_examples/automotive_soa/src/runtime.cpp

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "owl/runtime.hpp"
+
+namespace owl
+{
+Runtime& Runtime::GetInstance(const core::String& name) noexcept
+{
+    iox::runtime::PoshRuntime::initRuntime(iox::RuntimeName_t(iox::cxx::TruncateToCapacity, name));
```

Comment:
Not sure if your analysis is correct. Both methods were called but only one Runtime was created. Anyway I called the 2nd method from the first one, just to be sure and to avoid code duplication.